protocols/imap/src/main/java/org/apache/james/imap/processor/fetch/FetchProcessor.java

```diff
@@ -188,10 +188,10 @@ protected FetchGroup getFetchGroup(FetchData fetch) {
         FetchGroup result = FetchGroup.MINIMAL;
 
         if (fetch.isEnvelope()) {
-            result.or(FetchGroup.HEADERS_MASK);
```

Comment:
as far as I can tell, this class is not covered by tests, I thus don't agree with this refactoring.

Reply:
There are PLENTY of tests, MPT IMAP is doing an amazing job for fetch, every fetch level, on every implementation, and with a good message sample.

I feel comfortable! That might not be tests that you like but they are doing their job!

Reply:
I don't dislike MPT tests however I believe a lot in the "test pyramid" and we should not rely on high level tests to cover what unit tests should. The first reason is: you have no way to know that the test coverage is good enough whereas with unit test it's doable.

Reply:
Then... lets delay fetchGroup immutability, linked to FetchParser tests to another PR to not delay this changeset any longer.

Will contribute it tomorrow.